src/pages/index.js

```diff
@@ -0,0 +1,363 @@
+const profileDescription = document.querySelector(".profile__description");
+const avatarModalBtn = document.querySelector(".profile__avatar-btn");
+
+//3. Destructure the second item in the callback of the .then()
+api
+  .getAppInfo()
+  .then(([cards, user]) => {
+    cards.forEach((item) => {
+      const cardElement = getcardElement(item);
+      cardsList.prepend(cardElement);
+    });
+
+    setUserData(user);
+  })
+
+  .catch(console.error);
+
+// profile elements
+const profileEditButton = document.querySelector(".profile__edit-btn");
+const profileAvatar = document.querySelector(".profile__avatar");
+
+// form elements
+const editProfileModal = document.querySelector("#edit-modal");
+const editFormElement = document.forms["edit-profile"];
+const editButtonEl = editFormElement.querySelector(".modal__submit-btn");
+const editProfileModalClosedBtn = document.querySelector(".modal__close-btn");
+const editModalNameInput = editProfileModal.querySelector(
+  "#profile-name-input"
+);
+
+const editModalDescriptionInput = editProfileModal.querySelector(
+  "#profile-description-input"
+);
+
+// modal submit elements
+const cardSubmitModal = document.querySelector("#add-card-modal");
+const cardSubmitElement = cardSubmitModal.querySelector(".modal__submit-btn");
+const cardFormElement = document.forms["new-post"];
+const cardCloseBtn = cardSubmitModal.querySelector(".modal__close-btn");
+const cardNameInput = cardSubmitModal.querySelector("#add-card-name-input");
+const cardLinkInput = cardSubmitModal.querySelector("#add-card-link-input");
+
+// avatar form elements
+const avatarModal = document.querySelector("#avatar-modal");
+const avatarSubmitModal = document.querySelector("#avatar-modal");
+const avatarSubmitElement =
+  avatarSubmitModal.querySelector(".modal__submit-btn");
+const avatarFormElement = avatarModal.querySelector("#edit-avatar-form");
+const avatarCloseBtn = avatarSubmitModal.querySelector(".modal__close-btn");
+const avatarLinkInput = avatarSubmitModal.querySelector(
+  "#profile-avatar-input"
+);
+
+// delete form elements
+const deleteModal = document.querySelector("#delete-modal");
+const deleteForm = deleteModal.querySelector(".modal__form");
+
+// select the modal
+const previewModal = document.querySelector("#preview-modal");
+const previewModalImageEl = document.querySelector(".modal__image");
+const previewModalCaptionEl = document.querySelector(".modal__caption");
+const previewModalCloseBtn = previewModal.querySelector(".modal__close-btn");
+
+// card related elements
+const cardTemplate = document.querySelector("#card-template");
+const cardsList = document.querySelector(".cards__list");
+
+let selectedCard;
+let selectedCardId;
+
+function handleDeleteSumbit(evt) {
+  evt.preventDefault();
+  const button = evt.target.querySelector(".modal__submit-btn_delete");
+  api.deleteCard(selectedCardId).then(() => {
+    selectedCard.remove();
+    closeModal(deleteModal);
+  });
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**